gamemodes/sss/core/itemtype/hats.pwn

```diff
@@ -111,6 +111,13 @@ stock SetPlayerHatItem(playerid, Item:itemid)
 
 	RemoveItemFromWorld(itemid);
 	RemoveCurrentItem(GetItemHolder(itemid));
+	
+	if(hat_CurrentHatItem[playerid] == itemid)
```

Comment:
Shouldn't this be further up in the function, as a guard clause?

Reply:
Yes